src/main/java/picard/sam/markduplicates/UmiAwareDuplicateSetIterator.java

```diff
@@ -79,18 +88,20 @@
     public void close() {
         isOpen = false;
         wrappedIterator.close();
+
+        if (metrics.UMI_LENGTH > 0) {
```

Comment:
I don't understand why this if statement...what will happen if the UMI_LENGTH is 0?